dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowFactoryImpl.java

```diff
@@ -1073,6 +1118,293 @@ public void saveAction(final WorkflowAction workflowAction,
+			}
+		}
+
+		return mappingRow;
+	}
+
+	/**
+	 * Finds the list of {@link SystemActionWorkflowActionMapping}  associated to the {@link com.dotmarketing.portlets.workflows.business.WorkflowAPI.SystemAction}
+	 * and {@link List} of {@link WorkflowScheme}'s
+	 * @param systemAction {@link com.dotmarketing.portlets.workflows.business.WorkflowAPI.SystemAction}
+	 * @param schemes      {@link List} of {@link WorkflowScheme}'s
+	 * @return List<Map<String, Object>>
+	 */
+	@Override
+	public List<Map<String, Object>> findSystemActionsBySchemes(final WorkflowAPI.SystemAction systemAction, final List<WorkflowScheme> schemes) throws DotDataException {
+
+		final List<String> schemeIdList   = schemes.stream().map(WorkflowScheme::getId).collect(Collectors.toList());
+		List<Map<String, Object>> results = this.cache.findSystemActionsBySchemes(systemAction.name(), schemeIdList);
+
+		if (null == results) {
+
+			if (UtilMethods.isSet(schemes)) {
+
+				final DotConnect dotConnect = new DotConnect()
+						.setSQL(String.format(sql.SELECT_SYSTEM_ACTION_BY_SYSTEM_ACTION_AND_SCHEMES, this.createQueryIn(schemes)))
+						.addParam(systemAction.name());
+
+				for (final WorkflowScheme scheme : schemes) {
+
+					dotConnect.addParam(scheme.getId());
+				}
+
+				results = dotConnect.loadObjectResults();
+				if (UtilMethods.isSet(results)) {
+					this.cache.addSystemActionsBySystemActionNameAndSchemeIds(
+							systemAction.name(), schemeIdList, results);
+				}
+			}
+		}
+
+		return results;
+	}
+
+	@Override
+	public Map<String, Object> findSystemActionByIdentifier(final String identifier) throws DotDataException {
+
+		Map<String, Object> systemActionMap = cache.findSystemActionByIdentifier(identifier);
+
+		if (null == systemActionMap) {
+
+			final List<Map<String, Object>> rows = new DotConnect().setSQL(sql.SELECT_SYSTEM_ACTION_BY_IDENTIFIER)
+					.addParam(identifier)
+					.loadObjectResults();
+			systemActionMap = UtilMethods.isSet(rows) ? rows.get(0) : Collections.emptyMap();
+
+			this.cache.addSystemActionByIdentifier(identifier, systemActionMap);
+		}
+
+		return systemActionMap;
+	}
+
+	@Override
+	public boolean deleteSystemAction(final SystemActionWorkflowActionMapping mapping) throws DotDataException {
+
+		new DotConnect().setSQL(sql.DELETE_SYSTEM_ACTION_BY_IDENTIFIER)
+				.addParam(mapping.getIdentifier())
+				.loadResult();
+
+		this.cache.removeSystemActionByIdentifier(mapping.getIdentifier());
+
+		return true; //  todo: if rows affected 0 -> false
+	}
+
+	@Override
+	public void deleteSystemActionsByWorkflowAction(final WorkflowAction action) throws DotDataException {
+
+		Logger.debug(this, ()->"Removing system action mappings associated to the action: " + action.getId());
+		new DotConnect().setSQL(sql.DELETE_SYSTEM_ACTION_BY_WORKFLOW_ACTION_ID)
+				.addParam(action.getId())
+				.loadResult();
+
+		this.cache.removeSystemActionsByWorkflowAction(action.getId());
+	}
+
+	private String createQueryIn (final Collection list) {

```

Comment:
partition the list you are sending to the `where in` clause so we don't kill the dbs.

Reply:
Got it, will be fixed